juju/juju
# Remove stub firewallers

There is no need to implement `environs.Firewaller` with methods returning `nil` or "not supported" errors. Doing so is actually wasteful, because it prevents the uninstallation of the model-level firewaller worker.

Here we remove such implementations from the following providers:
- Oracle
- Manual
- Equinix

The interface is not currently implemented by the MAAS, LXD or VSphere providers, the latter making use of instance-level firewalling without issue.

## QA steps

We want to test that instance-firewalling still works without the environ firewaller implementation. I used Equinix for this, but OCI should do as well.
- Bootstrap, then `juju ssh` to the controller machine.
- `sudo iptables -L looks something like this:
```
$ sudo iptables -L
Chain INPUT (policy ACCEPT)
target     prot opt source               destination

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination
```
- Deploy an app to the controller, then expose it with `juju deploy ubuntu -m controller parca --to 0 && juju expose -m controller parca`.
- IP tables should now show a new Juju rule:
```
Chain INPUT (policy ACCEPT)
target     prot opt source               destination
ACCEPT     tcp  --  anywhere             anywhere             tcp dpt:7070 /* juju ingress */

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination
```

## Documentation changes

None.

## Bug reference

N/A
